components/tel/Core/Src/freertos.c

```diff
@@ -0,0 +1,462 @@
+    }
+
+    HAL_CAN_ActivateNotification(&hcan, CAN_IT_RX_FIFO0_MSG_PENDING);
+
+  }
+}
+
+__NO_RETURN void transmitMessageTask(void *argument) {
+  static CAN_msg_t can_message;
+  osStatus_t queue_status;
+
+  while (1) {
+    // retrieve CAN message from queue
+    queue_status = osMessageQueueGet(canMessageQueueHandle, &can_message, NULL, osWaitForever);
+
+    if (queue_status != osOK){
+      osThreadYield();
+    }
+
+    uint8_t can_buffer[21] = {0};
+
+    // TIMESTAMP: 8 ASCII characters
+    for (uint8_t i=0; i<8; i++) {
+      // send 'D' as placeholder
+        can_buffer[i] = 'D';
+    }
+
+    // CAN ID: 4 ASCII characters
+    uint8_t id_h = 0xFFUL & (can_message.header.StdId >> 8);
+    uint8_t id_l = 0xFFUL & (can_message.header.StdId);
+
+    can_buffer[8] = id_h;
+    can_buffer[9] = id_l;
+
+
+    // CAN DATA: 16 ASCII characters
+    for (uint8_t i=0; i<8; i++) {
+      // can_stream[2+i] = 0xFFUL & (can_message.data[i]);
+        can_buffer[i + 10]= can_message.data[i];
+    }
+
+    // CAN DATA LENGTH: 1 ASCII character
+    uint8_t length = "0123456789ABCDEF"[ can_message.header.DLC & 0xFUL];
+    can_buffer[18] = length;
+
+    // NEW LINE: 1 ASCII character
+    can_buffer[19] = '\n';
+
+    // CARRIAGE RETURN: 1 ASCII character
+    can_buffer[20] = '\r';
+
+    HAL_UART_Transmit(&huart3, can_buffer, sizeof(can_buffer), 1000);
+
+  }
+}
+
+__NO_RETURN void changeRadioSettingsTask(void *argument) {
+
+  while(1) {
+    // enter command mode
+    uint8_t tripleplus[3] = {'+', '+', '+'};
+
+    osDelay(1000);
+    HAL_UART_Transmit(&huart3, tripleplus, sizeof(tripleplus), 1000);
+    osDelay(2000);
+    
+    uint8_t check_dl[5] = "ATDL\r";
+    HAL_UART_Transmit(&huart3, check_dl, sizeof(check_dl), 1000);
+    
+    osDelay(1000);
+
+    // send destination low change AT command
+    uint8_t change_dl[15] = "ATDL 418C9CBD\r";
+    HAL_UART_Transmit(&huart3, change_dl, sizeof(change_dl), 1000);
+
+    osDelay(1000);
+
+    // send ATDL
+    HAL_UART_Transmit(&huart3, check_dl, sizeof(check_dl), 1000);
+    
+    osDelay(1000);
+    
+    // send "ATAC" (apply changes)
+    uint8_t apply_changes[5] = "ATAC\r";
+    HAL_UART_Transmit(&huart3, apply_changes, sizeof(apply_changes), 1000);
+
+    osDelay(1000);
+
+    // send "ATWR" (write changes to non-volatile memory)
+    uint8_t write_changes[5] = "ATWR\r";
+    HAL_UART_Transmit(&huart3, write_changes, sizeof(write_changes), 1000);
+
+    osDelay(1000);
+
+    HAL_GPIO_WritePin(KERNEL_LED_GPIO_Port, KERNEL_LED_Pin, GPIO_PIN_SET);
+
+    osThreadExit();
+  }
+}
+
+void sendChar(char c)
+{
+  HAL_StatusTypeDef uart_tx_status;
+
+  //Convert to ASCII
+  uint8_t c_H = "0123456789ABCDEF"[(c >> 4) & 0xFUL];
+  uint8_t c_L = "0123456789ABCDEF"[ c & 0xFUL];
+
+  uart_tx_status = HAL_UART_Transmit(&huart3, &c_H, 1, 1000);
+  uart_tx_status = HAL_UART_Transmit(&huart3, &c_L, 1, 1000);
+}
+
+__NO_RETURN void StartReadIMU(void *argument){
```

Comment:
As with function documentation, same for all tasks, short descriptions would be helpful.